aptos-move/block-executor/src/executor.rs

```diff
@@ -199,16 +165,18 @@ where
             }
         }
 
-        if last_input_output
-            .record(idx_to_execute, sync_view.take_reads(), result)
-            .is_err()
-        {
-            // When there is module publishing r/w intersection, can early halt BlockSTM to
-            // fallback to sequential execution.
-            scheduler.halt();
-            return SchedulerTask::NoTask;
-        }
+        let _ignore = last_input_output.record(idx_to_execute, sync_view.take_reads(), result);
         scheduler.finish_execution(idx_to_execute, incarnation, updates_outside)
+
+        // if !last_input_output.record(idx_to_execute, sync_view.take_reads(), result) {
```

Comment:
This only changes behavior when there are actual Module R/W conflicts, and after you tried it, the tests that hang were module r/w tests. So maybe here it was actually halting properly, but the other part has some problem?